### PR TITLE
Centralizing Pipeline Files, Removing PowerShell, Defaulting in Code, Prepping for Release Methodology

### PR DESCRIPTION
**Pull request by scbedd**
Creating devops specific tasks such that the processing of glob strings or the like are handled in one common location. `common_tasks.py` will also be where Environment Variable reads will occur for release pipeline packaging. 

Side benefit of the above is that the wonky PowerShell steps have been removed, and the build definition is much easier to parse as a result.

Also removing extraneous yaml files that are unnecessary with the correct YAML exploitation. 

#4182

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

**Comment by scbedd** [comment]
@Azure/azure-sdk-eng

File: .azure-pipelines/client.yml [hunk 1]
```diff
@@ -18,45 +18,113 @@ jobs:
...
+     pip install pathlib
+    displayName: 'Prep Environment'
 
-- template: generic.test.yml
-  parameters:
-    name: macOS
-    vmImage: 'macOS-10.13'
+  - script: |
+     cd ~/
+     wget https://bitbucket.org/pypy/pypy/downloads/pypy3-v6.0.0-linux64.tar.bz2
```
Comment (weshaggard) [inline .azure-pipelines/client.yml]: Is bitbucket the best place to download pypy?

Comment (scbedd) [reply .azure-pipelines/client.yml]: It is where their official site points to.

File: .azure-pipelines/client.yml [hunk 2]
```diff
@@ -18,45 +18,113 @@ jobs:
 
   - script: |
      pip install wheel
+     pip install setuptools
+     pip install pathlib
```
Comment (mikeharder) [inline .azure-pipelines/client.yml]: NIT: I think these 3 lines could be shortened to 1:

```
pip install wheel setuptools pathlib
```

File: .azure-pipelines/client.yml [hunk 3]
```diff
@@ -18,45 +18,113 @@ jobs:
...
+  - task: UsePythonVersion@0
+    displayName: 'Use Python $(python.version)'
+    inputs:
+      versionSpec: '$(python.version)'
+
+  - script: |
+     pip install wheel
+     pip install setuptools
+     pip install pathlib
+     pip install twine
```
Comment (mikeharder) [inline .azure-pipelines/client.yml]: NIT: Combine 4 lines into 1

File: .azure-pipelines/client.yml [hunk 4]
```diff
@@ -18,45 +18,113 @@ jobs:
...
+- job: Special_Python_Distro_Tests
+  dependsOn:
+  - 'Build_Target_Packages'
+  pool:
     vmImage: 'ubuntu-16.04'
+  steps:
+  - script: |
+     pip install wheel
+     pip install setuptools
+     pip install pathlib
```
Comment (mikeharder) [inline .azure-pipelines/client.yml]: NIT: Combine 3 lines into 1

**Pull request closed without merging**